server/enhancedDate.js

```diff
@@ -1 +1,72 @@
-// export this as a Node module using the enhancedDate.js code from Module_2
\ No newline at end of file
+// export this as a Node module using the enhancedDate.js code from Module_2
+var _date;
+var weekday = ["Sunday", "Monday", "Tuesday", "Wednesday", "Thursday", "Friday", "Saturday"];
+var month = ["January", "February", "March", "April", "May", "June", "July", "Auguest", "September", "October", "Novenber", "December"];
+var current = new Date();
+var dateIsSet = false;
+
+function _setDate(oDate) {
+	if(oDate instanceof Date) {
+		_date = oDate;
+	} else {
+		_date = current;
+	}
+
+	dateIsSet = true;
+}
+function _getDate(oDate) {
+	if(!dateIsSet) {
+		_setDate();
+	}
+	if(oDate === true) {
+		return _date;
+	} else {
+		return _date.getTime();
+	}
+}
+function _getDayName() {
+	if(!dateIsSet) {
+		_setDate();
+	}
+	return weekday[_date.getDay()];
+}
+function _getMonthName() {
+	if(!dateIsSet) {
+		_setDate();
+	}
+	return month[_date.getMonth()];
+}
+function _isFuture() {
+	if(!dateIsSet) {
+		_setDate();
+	}
+	if( _date.getDay() >= current.getDay() ) {
```

Comment:
getDay returns a number from 0 to 6 that corresponds to a day of the week - this is insufficient to determine if _date is in the future or not relative to the current time.